docs/bundles/plain.md

```diff
@@ -2,43 +2,147 @@
 
 ## Overview
 
-This document is meant to define the `plain` bundle format as a reference for those publishing `plain` bundles
-for use with RukPak. For more information on the concept of a bundle, click [here](https://github.com/operator-framework/rukpak#bundle).
+This document is meant to define the plain bundle format as a reference for those publishing plain bundles for use with
+RukPak. A bundle is a collection of Kubernetes resources that are packaged together for the purposes of installing onto
+a Kubernetes cluster. A bundle can be unpacked onto a running cluster, where controllers can then create the underlying
+content embedded in the bundle. The bundle can be used as the underlying `spec.source` for
+a [`Bundle`](https://github.com/operator-framework/rukpak#bundle) resource.
+
+A plain bundle is a static collection of arbitrary Kubernetes YAML manifests. A plain bundle is contained in a directory
+that can be packaged in a container image layer , a `git` repository or any other content source that the
+[plain bundle provisioner](../provisioners/plain.md) supports.
+
+Supported source types for a plain bundle currently include the following:
+
+* A directory in a [container image](../sources/image.md)
+* A directory in a [`git` repository](../sources/git.md)
+* A set of keys in a [`ConfigMap`](../sources/local.md)
+* An [upload](../sources/upload.md)
+* A `.tgz` file returned by a [http endpoint](../sources/http.md)
 
-A plain bundle is simply a collection of static, arbitrary, Kubernetes YAML manifests in a given directory.
 
 The currently implemented plain bundle format is the `plain+v0` format. The name of the bundle format, `plain+v0`
-combines the type of bundle (plain) with the current schema version (v0).
+combines the type of bundle (plain) with the current schema version (v0). The
+[plain bundle provisioner](../provisioners/plain.md) is able to source
+`plain+v0` bundles and install them onto a Kubernetes cluster.
 
-> Note: the plain+v0 bundle format is at schema version v0, which means it's an experimental format that is subject
+> Note: the `plain+v0` bundle format is at schema version v0, which means it's an experimental format that is subject
 > to change.
 
-## Example
+## Common Terminology
+
+* *bundle* is a collection of files that define content to be deployed to a cluster
+* *bundle image* is a container image that contains a bundle within its filesystem
+* *bundle git repo* is a git repository that contains a bundle within a directory
+
+## Technical Details
+
+* The static manifests must be located in the root-level `/manifests` directory for the bundle to be a
```

Comment:
Here it is the "image" or "the git repo", that has a `/manifests`directory, which is a called a bundle. I prefer this but it is not consistent with what is described above.

Reply:
I believe this is standard for all but the `ConfigMap` source type, but I could be wrong. 

Would you mind elaborating on how it is inconsistent with what is described above?

Reply:
> Would you mind elaborating on how it is inconsistent with what is described above?

You write "the static manifests must be located in the root-level `/manifests` directory for the bundle to be a valid `plain+v0` bundle that the provisioner can unpack."
- That the location matters for the validity of "a collection of manifests" does not sound right. It does sound right if you refer to an image containing files.
- When you write that the provisioner can **unpack** a bundle you are referring to an image. "Unpacking" cannot really be applied to "a collection of manifests".

 To keep "the collection of manifests" concept I would rather expect something like: 
 "the static manifests must be located in the root-level `/manifests` directory for the provisioner to be able to locate the `plain+v0` bundle inside an OCI image, git repository or ConfigMap.
[ 
 According to the API ](https://github.com/operator-framework/rukpak/blob/main/api/v1alpha1/bundle_types.go#L80-L109)it seems however that it is only true for an OCI Image as it is possible to specify a directory for a Git repo and a path for a ConfigMap. Although, I am not following
 - why the same option is not offered for an OCI image
 - why it is called `directory`for a Git repo and `path`for a ConfigMap

Reply:
Ah, I see. Thanks for elaborating - that definitely clarified it for me and I agree that we should go with something more like:
>"the static manifests must be located in the root-level /manifests directory for the provisioner to be able to locate the plain+v0 bundle inside an OCI image, git repository or ConfigMap."

Although since the goal of this doc is to explain what a plain bundle is rather than focusing on provisioner details we should probably refine this a bit more in a way that doesn't mention the provisioner. I'll attempt to reword this section a bit.

Reply:
I couldn't come up with a good way to remove the mentioning of the provisioner, but I did reword it in https://github.com/operator-framework/rukpak/pull/642/commits/d2f0fc91f91ca25a2623fe241adbe33897b00629 to state that the plain bundle provisioner is opinionated and expects the plain bundle to be located in the `/manifests` directory.

Let me know if this addresses your concern regarding consistency

Reply:
Yes it does address my concern regarding consistency. I think that "is opinionated" is redundant but it is nitpicking. My impression is that the fact that only /manifests is supported is an implementation restriction rather than a design decision. Especially when it is not the case for Git repository and ConfigMaps.

Reply:
>My impression is that the fact that only /manifests is supported is an implementation restriction rather than a design decision. Especially when it is not the case for Git repository and ConfigMaps.

I think this is a fair impression and something I agree should be addressed but I would say is out of scope for this particular PR. I noticed you already filed an issue (https://github.com/operator-framework/rukpak/issues/653) to discuss this so I'm going to mark this comment as resolved. Feel free to unresolve if you feel it hasn't properly been resolved